x-pack/plugins/security_solution/common/endpoint/index_data.ts

```diff
@@ -98,3 +198,266 @@ async function indexAlerts(
+    ],
+    namespace: 'default',
+    package: {
+      name: 'endpoint',
+      title: 'endpoint',
+      version: endpointPackageVersion,
+    },
+  };
+  const packagePolicy = (await kbnClient.request({
+    path: PACKAGE_POLICY_API_ROUTES.CREATE_PATTERN,
+    method: 'POST',
+    body: newPackagePolicyData,
+  })) as AxiosResponse<CreatePackagePolicyResponse>;
+  return packagePolicy.data.item;
+};
+
+const getEndpointPackageInfo = async (
+  kbnClient: KbnClient
+): Promise<GetPackagesResponse['response'][0]> => {
+  const endpointPackage = ((await kbnClient.request({
+    path: `${EPM_API_ROUTES.LIST_PATTERN}?category=security`,
+    method: 'GET',
+  })) as AxiosResponse<GetPackagesResponse>).data.response.find(
+    (epmPackage) => epmPackage.name === 'endpoint'
+  );
+
+  if (!endpointPackage) {
+    throw new Error('EPM Endpoint package was not found!');
+  }
+
+  return endpointPackage;
+};
+
+const fleetEnrollAgentForHost = async (
+  kbnClient: KbnClient & {
+    requestWithApiKey: (path: string, init?: RequestInit | undefined) => Promise<Response>;
+  },
+  endpointHost: HostMetadata,
+  agentPolicyId: string
+): Promise<undefined | PostAgentEnrollResponse['item']> => {
+  // Get Enrollement key for host's applied policy
+  const enrollmentApiKey = await kbnClient
+    .request<GetEnrollmentAPIKeysResponse>({
+      path: ENROLLMENT_API_KEY_ROUTES.LIST_PATTERN,
+      method: 'GET',
+      query: {
+        kuery: `fleet-enrollment-api-keys.policy_id:"${agentPolicyId}"`,
+      },
+    })
+    .then((apiKeysResponse) => {
+      const apiKey = apiKeysResponse.data.list[0];
+
+      if (!apiKey) {
+        return Promise.reject(
+          new Error(`no API enrolment key found for agent policy id ${agentPolicyId}`)
+        );
+      }
+
+      return kbnClient.request<GetOneEnrollmentAPIKeyResponse>({
+        path: ENROLLMENT_API_KEY_ROUTES.INFO_PATTERN.replace('{keyId}', apiKey.id),
+        method: 'GET',
+      });
+    })
+    .then((apiKeyDetailsResponse) => {
+      return apiKeyDetailsResponse.data.item.api_key;
+    })
+    .catch((error) => {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      return '';
+    });
+
+  if (enrollmentApiKey.length === 0) {
+    return;
+  }
+
+  // Enroll an agent for the Host
+  const body: PostAgentEnrollRequest['body'] = {
+    type: 'PERMANENT',
+    metadata: {
+      local: {
+        host: endpointHost.host,
+        elastic: {
+          agent: {
+            version: '8.0.0',
+          },
+        },
+      },
+      user_provided: {
+        dev_agent_version: '0.0.1',
+        region: 'us-east',
+      },
+    },
+  };
+
+  try {
+    // FIXME: should we use `fetch()` in this module?
+    // FIXME: need way to get kibana URL without auth in it
+
+    // ------------------------------------------------
+    // First enroll the agent
+    const res = await kbnClient.requestWithApiKey(
+      `http://localhost:5601${AGENT_API_ROUTES.ENROLL_PATTERN}`,
```

Comment:
Here and anywhere you call `.requestWithApiKey()` - remove the full url and only pass in the path:

```sugestion
      AGENT_API_ROUTES.ENROLL_PATTERN,
```

the method `.requestWithApiKey()` will append the path to the kibana URL.

Reply:
kk